atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTableDropper.java

```diff
@@ -39,68 +38,68 @@
+        this.cassandraTableTruncator = cassandraTableTruncator;
     }
 
     void dropTables(final Set<TableReference> tablesToDrop) {
         try {
-            clientPool.runWithRetry(
-                    (FunctionCheckedException<CassandraClient, Void, Exception>) client -> {
-                        KsDef ks = client.describe_keyspace(
-                                config.getKeyspaceOrThrow());
-                        Set<TableReference> existingTables = Sets.newHashSet();
+            clientPool.runWithRetry(client -> {
+                KsDef ks = client.describe_keyspace(config.getKeyspaceOrThrow());
+                Set<TableReference> existingTables = Sets.newHashSet();
 
-                        existingTables.addAll(ks.getCf_defs().stream()
-                                .map(CassandraKeyValueServices::tableReferenceFromCfDef)
-                                .collect(Collectors.toList()));
+                existingTables.addAll(ks.getCf_defs().stream()
+                        .map(CassandraKeyValueServices::tableReferenceFromCfDef)
+                        .collect(Collectors.toList()));
 
-                        for (TableReference table : tablesToDrop) {
-                            CassandraVerifier.sanityCheckTableName(table);
-                            if (existingTables.contains(table)) {
-                                client.system_drop_column_family(
-                                        CassandraKeyValueServiceImpl.internalTableName(table));
-                                putMetadataWithoutChangingSettings(table,
-                                        AtlasDbConstants.EMPTY_TABLE_METADATA);
-                            } else {
-                                log.warn("Ignored call to drop a table ({}) that did not exist.",
-                                        LoggingArgs.tableRef(table));
-                            }
-                        }
-                        CassandraKeyValueServices.waitForSchemaVersions(config, client, "after dropping the column "
-                                + "family for tables " + tablesToDrop + " in a call to drop tables");
-                        return null;
-                    });
+                for (TableReference table : tablesToDrop) {
+                    CassandraVerifier.sanityCheckTableName(table);
+                    if (existingTables.contains(table)) {
+                        cassandraTableTruncator.runTruncateOnClient(ImmutableSet.of(table), client);
+                        client.system_drop_column_family(CassandraKeyValueServiceImpl.internalTableName(table));
+                        updateMetadataForTable(table);
+                    } else {
+                        log.warn("Ignored call to drop a table ({}) that did not exist.", LoggingArgs.tableRef(table));
+                    }
+                }
+                CassandraKeyValueServices.waitForSchemaVersions(config, client, "after dropping the column "
```

Comment:
I wonder if we should be wrapping every drop/create call with this (both sides of it and at the individual resource level). So something more like:
```
for (TableReference table: tablesToDrop) {
    // wait for schema versions
    // drop table
    // wait for schema versions
}
```